### PR TITLE
List user references

### PR DESCRIPTION
**Pull request by mrkvon**
#### Proposed Changes

* Read references of user.
React component.

`/profile/{username}/references`
- [x] show list of references which user received
  - [x] name, username, avatar, interactions, recommendation, when the reference was given
  - [ ] design
- [ ] when looking at my references, show which ones I haven't given back (public or private) and how much time is left till they become public. Provide a button.
  - [x] show unpublished references to me
    - [x] avatar, link, time left to give
    - [ ] design
- [ ] show list of references which user gave
- [ ] when looking at my references, show which ones I gave and they're still private (and time left)

#### Testing Instructions

* manual testing

Continues fixing #718 
Depends on #827 
It will need a UX and looks design.

#### To do
- [ ] show a number of references in user tab (separated by recommend (yes,no,unknown))
- [ ] show a list of references when the tab is clicked

### Screenshots

05_list_references_self

### CODE REVIEW

**Comment by simison** [comment]
FYI, I'm looking into designs for this.

**Comment by petr-hajek** [comment]
> FYI, I'm looking into designs for this.

Still looking into it? Or I could try to make something up.

**Comment by simison** [comment]
>  Still looking into it? Or I could try to make something up.

Still actively thinking it but haven't actually started plotting any designs yet. Happy if you get started with it and we can iterate on it together!

**Comment by petr-hajek** [comment]
> > Still looking into it? Or I could try to make something up.
> 
> @petr-hajek I'm still actively thinking it but haven't actually started plotting any designs yet. Happy if you get started with it and we can iterate on it together!

I will be going out of the collective soon, so I wanna wrap things up. I'll be probably offline for longer period now.

**Comment by simison** [comment]
This PR will need same shallow `setState` changes as https://github.com/Trustroots/trustroots/pull/827

It's okay if we first get very rough version of this merged. We can then work out the UI / design out before publishing the feature.

**Comment by mrkvon** [comment]
A raw displaying of references on the reference page is more or less done. We'll need a design to progress further.

**Comment by mrkvon** [comment]
Hello there! I need a design or a code review by someone to progress on this further.
I'll self-review today.
Merge without design would also be an option (and then it may stay hidden behind flags and be tested and iterated upon on a dev server or something (dev server is down though)).

**Comment by simison** [comment]
@mrkvon thanks for the nudge. I'd say go ahead with self-review, I'll loop back to design aspects.

Let's chat more later about speeding up the process and freeing you up from reviews a bit, going forward. :-)

**Comment by simison** [comment]
Rebased! :tada:

**Comment by simison** [comment]
Could someone take screengrabs of the feature as it is currently working here? It would be helpful for me to see and give feedback.

**Comment by mrkvon** [comment]
> take screengrabs

@simison a screenshot is in the PR description

Additional screenshots (create a reference) will appear in #718 shortly.

Also rebased.

**Comment by mrkvon** [comment]
Rebased and fixed conflicts. Anyways, this is not in my focus, currently. Feel free to close or take over this PR.

**Comment by mrkvon** [comment]
Rebased this with @edvardmk and @rumwerfer. Also refactored a little.

Who wants to implement listing references? This PR is for grabs.

I'm ready to answer questions about this PR and give support to whoever may want to continue it.
Alternatively, if [somebody provides a clear specification based on consensus and at least a basic user interface design] and [nobody else is interested in working on this] and [this is needed], then i may implement it myself.

Also depending on what we want it may be easier to close this and start from scratch.

**Comment by Akronix** [comment]
hey @mrkvon I'm up to take some time slots of my agenda this week and work to polish this.

File: modules/core/client/components/Loading.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+/**
+ * I'm just a little loading info.
+ */
+export default function Loading() {
```
Comment (simison) [inline modules/core/client/components/Loading.js:7]: Let's use existing component:

https://github.com/Trustroots/trustroots/blob/master/modules/core/client/components/LoadingIndicator.js

File: modules/references/client/components/create-reference/Recommend.js [hunk 2]
```diff
@@ -17,13 +17,15 @@ export default function Recommend({
...
-    hostedThem: t('Would you recommend others to host them?'),
-    met: t('Would you recommend others to meet them?'),
+    hostedMe: t('Did you feel comfortable being hosted by them?'),
+    hostedThem: t('Did you feel comfortable hosting them?'),
+    met: t('Did you feel comfortable meeting them?'),
   };
 
   const question = recommendQuestions[primaryInteraction];
 
+  const marginRight = { marginRight: '20px' };
```
Comment (simison) [inline modules/references/client/components/create-reference/Recommend.js:27]: Let's use either styled components in JS or CSS in less files, but no inline CSS via JS.

Comment (Akronix) [reply modules/references/client/components/create-reference/Recommend.js:27]: OK. The plan is to change upgrade react-bootstrap lib anyway and use different styling options from the new version.

**Comment by mrkvon** [comment]
I'd be in favor of opening a separate pull request based on the references-private-experience branch and merging that to master. We don't need to care about (or merge) all the #918 stuff when we want the UX changes in reference creating experience. That is, the recent changes by @Akronix and @rumwerfer.

**Comment by Akronix** [comment]
I agree. Let's close this to avoid further confusion

File: modules/references/client/components/create-reference/Recommend.js (hunk 2 above)
Comment (Akronix) [reply modules/references/client/components/create-reference/Recommend.js:27]: Is it resolved now?

**Comment by nicksellen** [comment]
^ I just restored the branch in case there is some useful code in there to reuse for the eventual branch we create...

**Pull request closed without merging**